storage/objects.go

```diff
@@ -258,6 +260,21 @@ type PutObjectInput struct {
 func (s *ObjectsClient) Put(ctx context.Context, input *PutObjectInput) error {
 	fullPath := path.Clean(path.Join("/", s.client.AccountName, input.ObjectPath))
 
+	err := checkDirectoryTreeExists(*s, ctx, input.ObjectPath)
```

Comment:
If we're not checking within the loop how are we handling existing directories?

Reply:
Demo'ed to @cheapRoc that we have a test case for this to work and works as expected